chain/chunks/src/shards_manager_actor.rs

```diff
@@ -1385,7 +1394,11 @@ impl ShardsManagerActor {
             }
         };
 
-        if !self.epoch_manager.verify_chunk_header_signature(header, &epoch_id, &ancestor_hash)? {
+        if !verify_chunk_header_signature_with_epoch_manager(
```

Comment:
@pugachAG I think the reason `ultra_slow_test_chunks_produced_and_distributed_one_val_shard_cop` is [failing](https://nayduck.nearone.org/#/test/252556) is because I screwed the `epoch_id` here

Reply:
It is not a random occurrence as it does not fail on master so I changed the semantics accidentally

Reply:
I tested further after the fixing the `epoch_id` issue and the test passes more often but it is still flaky. 

I may be going down the wrong rabbit hole, but I do not really see a difference in semantics apart from the fact that before the change, the epoch manager stayed locked for a longer period of time compared to now where it is released before calling `verify_chunk_header_signature`